lmfit/minimizer.py

```diff
@@ -887,6 +875,61 @@ def emcee(self, params=None, steps=1000, nwalkers=100, burn=0, thin=1,
 
         return result
 
+    def least_squares(self, params=None, **kws):
+        """
+        Use the least_squares (new in scipy 0.17) function to perform a fit.
+        This assumes that ModelParameters have been stored, and a function to
+        minimize has been properly set up. This method doesn't support
+        parameter.constraints.
+
+        This wraps scipy.optimize.least_squares, which has in build support
+        for bounds and robust loss functions.
+
+        Parameters
+        ----------
+        params : Parameters, optional
+           Parameters to use as starting points.
+        kws : dict, optional
+            Minimizer options to pass to scipy.optimize.least_squares.
+
+        """
+
+        if not HAS_NEW_LEAST_SQUARES_MIN:
+            raise NotImplementedError("Scipy with a Version higher than 0.17 "
+                                      "is need for this method.")
+
+        result = self.prepare_fit(params)
+
+        replace_none = lambda x, sign: sign*np.inf if x is None else x
+        upper_bounds = [replace_none(i.max, 1) for i in self.params.values()]
+        lower_bounds = [replace_none(i.min, -1) for i in self.params.values()]
+        start_vals = [i.value for i in self.params.values()]
+
+        ret = least_squares(self.__residual,
```

Comment:
I'd be tempted to use `functools.partial`:

    resid = partial(self.__residual, apply_scaling=False)
    ret = least_squares(resid, ...)